x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/Div.java

```diff
@@ -63,21 +64,34 @@ public ArithmeticOperationFactory binaryComparisonInverse() {
 
     @Evaluator(extraName = "Ints", warnExceptions = { ArithmeticException.class })
     static int processInts(int lhs, int rhs) {
+        if (rhs == 0) {
+            throw new ArithmeticException("/ by zero");
+        }
         return lhs / rhs;
     }
 
     @Evaluator(extraName = "Longs", warnExceptions = { ArithmeticException.class })
     static long processLongs(long lhs, long rhs) {
+        if (rhs == 0L) {
+            throw new ArithmeticException("/ by zero");
+        }
         return lhs / rhs;
     }
 
     @Evaluator(extraName = "UnsignedLongs", warnExceptions = { ArithmeticException.class })
     static long processUnsignedLongs(long lhs, long rhs) {
+        if (rhs == NumericUtils.ZERO_AS_UNSIGNED_LONG) {
+            throw new ArithmeticException("/ by zero");
+        }
         return longToUnsignedLong(Long.divideUnsigned(longToUnsignedLong(lhs, true), longToUnsignedLong(rhs, true)), true);
     }
 
-    @Evaluator(extraName = "Doubles")
+    @Evaluator(extraName = "Doubles", warnExceptions = { ArithmeticException.class })
     static double processDoubles(double lhs, double rhs) {
-        return lhs / rhs;
+        double value = lhs / rhs;
+        if (Double.isNaN(value) || Double.isInfinite(value)) {
+            throw new ArithmeticException("/ by zero");
```

Comment:
This can also be due to overflow, so the error message is not always correct.

Why not just use `Math.divideExact` (also for the others except for unsigned longs)?

Same for `Mod`